test/e2e/prometheus_test.go

```diff
@@ -84,3 +85,98 @@ func TestPrometheusVolumeClaim(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestPrometheusAlertmanagerAntiAffinity(t *testing.T) {
+	pods, err := f.KubeClient.CoreV1().Pods(f.Ns).List(metav1.ListOptions{FieldSelector: "status.phase=Running"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var alm = `affinity:
+	   podAntiAffinity:
+	      preferredDuringSchedulingIgnoredDuringExecution:
+	      - podAffinityTerm:
+	          labelSelector:
+	            matchExpressions:
+	            - key: alertmanager
+	              operator: In
+	              values:
+	              - main`
+
+	var k8s = `affinity:
+	   podAntiAffinity:
+	      preferredDuringSchedulingIgnoredDuringExecution:
+	      - podAffinityTerm:
+	          labelSelector:
+	            matchExpressions:
+	            - key: prometheus
+	              operator: In
+	              values:
+	              - k8s`
+
+	var testPod1 = "alertmanager-main"
+	var testPod2 = "prometheus-k8s"
+	var testNameSpace = "openshift-monitoring"
+
+	mapPodAntiAff := map[int]string{0: ""}
```

Comment:
let's use native slices here instead of maps. we don't need this data structure here.

Reply:
correction, we don't need a slice/map whatsoever, see https://github.com/openshift/cluster-monitoring-operator/pull/476/files#r325243641.

Reply:
fixed